docs/reference/modules/asset.md

```diff
@@ -0,0 +1,219 @@
+---
+extends: '@apostrophecms/module'
+---
+
+# `@apostrophecms/asset`
+
+**Alias:** `apos.asset`
+
+<AposRefExtends :module="$frontmatter.extends" />
+
+The `asset` module serves to organize, process, and output all project JavaScript and CSS assets during the build process. In addition, it provides access to modify the project webpack configuration and exposes two CLI tasks for project building and webpack cache clearing. Options are passed through the creation of a `modules/@apostrophecms/asset/index.js` file.
+
+## Options
+
+|  Property | Type | Description |
+|---|---|---|
+| [`refreshOnRestart`](#refreshonrestart) | Boolean | If set to `true`, the browser will refresh on Apostrophe app restart. |
+| [`watch`](#watch) | Boolean | If set to `false`, none of the UI assets will be watched to trigger a restart. |
+| `watchDebounceMs` | Integer | Time in milliseconds to wait before re-triggering a restart on asset change. |
+| [`uploadfs`](#uploadfs) | String \|\| Object | Determines what processor to use for assets during the build process.
+
+### `refreshOnRestart`
+
+By default, `refreshOnRestart` is set to `false`. If this option has a falsy value, the browser will not automatically refresh following app restart.  If this option is truthy and `process.env.NODE_ENV` is not set to `production`, then restart of the Apostrophe app will trigger a browser refresh. This is very useful in combination with `nodemon` to deliver file changes to the browser.
+
+### `watch`
+
+By default, `watch` is set to `true`. A truthy value will cause the application to monitor for changes in any of the asset bundles in the `modules` and `node_modules` folders and trigger a webpack rebuild and browser refresh. A value of `false` will disable this behavior. Note that this option is disabled and has no impact if `process.env.NODE_ENV` is set to `production`.
+
+### `uploadfs`
+
+By default, uploadfs will use [sharp](https://www.npmjs.com/package/sharp) for static asset storage and delivery during the build process and during upload by the `attachment` module. Setting the `image` subproperty to `imagemagick` will force use of this processor if it is installed for the build process. Finally, a fully custom processor can be used by passing an object.  The processor choice for asset upload can be set seperately in the `attachment` module options. Full documentation for uploadfs can be found [here](https://www.npmjs.com/package/uploadfs).
```

Comment:
If we're going to go there and discuss the details of this option's subproperties here then it shouldn't sound like it's all about image processing. But it might be better to just say that this option can optionally be used to configure uploadfs differently from the instance used for uploaded media storage, in which case the properties of this object are passed to uploadfs [link to docs], allowing control of where files are stored and how they are served.

Image processing actually doesn't come into play at all for assets stored by the asset module, because it never calls `copyImageIn`. Apostrophe only does that for attachments.

Reply:
Changed. Okay - I think I get it. Longer form observation in the slack.